src/processor/operator/update/set_executor.cpp

```diff
@@ -0,0 +1,68 @@
+#include "processor/operator/update/set_executor.h"
+
+namespace kuzu {
+namespace processor {
+
+void NodeSetExecutor::init(ResultSet* resultSet, ExecutionContext* context) {
+    nodeIDVector = resultSet->getValueVector(nodeIDPos).get();
+    evaluator->init(*resultSet, context->memoryManager);
+    rhsVector = evaluator->resultVector.get();
+}
+
+void SingleLabelNodeSetExecutor::set() {
+    evaluator->evaluate();
+    for (auto i = 0u; i < nodeIDVector->state->selVector->selectedSize; ++i) {
+        auto lhsPos = nodeIDVector->state->selVector->selectedPositions[i];
+        auto& nodeID = nodeIDVector->getValue<common::internalID_t>(lhsPos);
+        auto rhsPos = lhsPos;
+        if (rhsVector->state->selVector->selectedSize == 1) {
+            rhsPos = 0;
+        }
+        column->write(nodeID.offset, rhsVector, rhsPos);
+    }
+}
+
+void MultiLabelNodeSetExecutor::set() {
+    evaluator->evaluate();
+    for (auto i = 0u; i < nodeIDVector->state->selVector->selectedSize; ++i) {
+        auto lhsPos = nodeIDVector->state->selVector->selectedPositions[i];
+        auto& nodeID = nodeIDVector->getValue<common::internalID_t>(lhsPos);
+        if (!tableIDToColumn.contains(nodeID.tableID)) {
+            continue;
+        }
+        auto column = tableIDToColumn.at(nodeID.tableID);
+        auto rhsPos = lhsPos;
+        if (rhsVector->state->selVector->selectedSize == 1) {
```

Comment:
This logic is quite implicit and if we handle the flat/unflat here, it seems to be duplicated with what we do in `Column::write(ValueVector* nodeIDVector, ValueVector* vectorToWriteFrom)`.

Here are two alternative solutions in my mind:
1. Manually flatten nodeIDVector here, so we can reuse `Column::write(ValueVector* nodeIDVector, ValueVector* vectorToWriteFrom)`.
2. Add a new interface `Column::write(offset_t nodeOffset, ValueVector* vectorToWriteFrom)`, which can handle both cases of `vectorToWriteFrom` being flat or unflat.